config/application.rb

```diff
@@ -128,5 +129,15 @@ class Application < Rails::Application
 
     # Do not swallow errors in after_commit/after_rollback callbacks.
     # config.active_record.raise_in_transactional_callbacks = true
+
+    # Protect from "invalid byte sequence in UTF-8" errors,
+    # when a query or a cookie is a string with incorrect UTF-8 encoding.
+    config.middleware.insert_before(
+      Rack::Head,
+      Rack::UTF8Sanitizer,
```

Comment:
I'm not familiar with adding custom middleware, so I'm just 100% going from the [docs](https://github.com/whitequark/rack-utf8_sanitizer?tab=readme-ov-file#strategies) and they seem to recommend a different structure to the code you've got here.  

Specifically it says `Note: The exception handling middleware must be inserted before Rack::UTF8Sanitizer`